sdks/java/build-tools/src/main/resources/beam/checkstyle/checkstyle.xml

```diff
@@ -119,6 +119,14 @@ page at http://checkstyle.sourceforge.net/config.html -->
       <property name="message" value="You are using raw guava, please use vendored guava classes."/>
     </module>
 
+    <!-- Forbid guava Charset. -->
```

Comment:
tested that the enforcement is effective

without source code changes, build fails with

```
> Task :sdks:java:core:checkstyleMain
[ant:checkstyle] [ERROR] /Users/yathu/beam/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/LoggingTransformProvider.java:39: You are using guava Charsets, please use Java builtin StandardCharsets. [ForbidGuavaCharset]
...


```